### PR TITLE
Calypso Codemods: Pull in to monorepo

### PR DESCRIPTION
**Pull request by sirreal**
#### Changes proposed in this Pull Request

* Adds `calypso-codemods` to the monorepo

#### Testing instructions

* npm ci
* Try codemods like:
* `./node_modules/.bin/calypso-codemods react-create-class client/components/`
* `npm run test-packages calypso-codemods`

### CODE REVIEW

**Comment by sirreal** [comment]
Rebased and lint-free 🤞

**Comment by sirreal** [comment]
Rebased and squashed Calypso prep commits.

**Comment by ockham** [comment]
Gonna rebase.

**Comment by ockham** [comment]
Oops, we renamed `npm-shrinkwrap.json` to `package-lock.json` two days ago (#37493). I'll amend the last commit...

File: packages/calypso-codemods/README.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
+# Calypso Codemods
+
+[![Build Status](https://travis-ci.org/Automattic/calypso-codemods.svg?branch=master)](https://travis-ci.org/Automattic/calypso-codemods)
```
Comment (ockham) [inline packages/calypso-codemods/README.md]: We'll need to remove, or update this :thinking:

File: packages/calypso-codemods/.eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+module.exports = {
+	rules: {
+		'import/no-nodejs-modules': 0,
```
Comment (ockham) [inline packages/calypso-codemods/.eslintrc.js]:
```suggestion
		'import/no-nodejs-modules': 'off',
```

File: packages/calypso-codemods/package.json [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+	},
+	"scripts": {
+		"test": "jest"
+	},
+	"repository": {
+		"type": "git",
+		"url": "git+https://github.com/Automattic/wp-calypso.git",
+		"directory": "packages/calypso-codemods"
+	},
+	"author": "Jake Fried <samouri@users.noreply.github.com>",
```
Comment (ockham) [inline packages/calypso-codemods/package.json]: Should be

```suggestion
	"author": "Automattic Inc.",
```

per #36534

File: packages/calypso-codemods/package.json [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+	"license": "GPL-2.0-or-later",
+	"bugs": {
+		"url": "https://github.com/Automattic/calypso-codemods/issues"
```
Comment (ockham) [inline packages/calypso-codemods/package.json]:
```suggestion
		"url": "https://github.com/Automattic/wp-calypso/issues"
```

File: packages/calypso-codemods/package.json [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+	},
+	"homepage": "https://github.com/Automattic/calypso-codemods#readme",
```
Comment (ockham) [inline packages/calypso-codemods/package.json]:
```suggestion
	"homepage": "https://github.com/Automattic/wp-calypso/tree/master/packages/calypso-codemods#readme",
```

**Review by ockham — APPROVED** [review]
Working well, thanks for bringing this into the fold!

Left a few suggestions which should be near-trivial to apply. Let's merge this soon and retire the separate repo!

**Comment by sirreal** [comment]
Thanks for the review! All concerns addressed.

**Pull request merged**